pkg/log/log.go

```diff
@@ -197,44 +199,72 @@ type MyFormatter struct{}
 
 func (mf *MyFormatter) Format(entry *logrus.Entry) ([]byte, error) {
 	var b *bytes.Buffer
+	var writeString string
 	if entry.Buffer != nil {
 		b = entry.Buffer
 	} else {
 		b = &bytes.Buffer{}
 	}
 	level := strings.ToUpper(entry.Level.String())
-	strList := strings.Split(entry.Caller.File, "/")
-	fileName := strList[len(strList)-1]
 	funcList := strings.Split(entry.Caller.Function, "/")
 	funcName := funcList[len(funcList)-1]
 	subIndex := strings.Index(funcName, ".")
 	if subIndex != -1 {
 		funcName = funcName[subIndex+1:]
 	}
 
-	b.WriteString(fmt.Sprintf("%s:[%s %s:#%d]  %s\n",
-		entry.Time.Format("2006-01-02 15:04:05 -0700"), level, fileName, entry.Caller.Line,
-		entry.Message))
+	// TODO: Once we migrate to ginkgo v2, we can extract test name as follows:
+	// report := CurrentSpecReport()
+	// testName := report.ContainerHierarchyTexts[0]
+	report := CurrentGinkgoTestDescription()
+	testName := strings.Split(report.FullTestText, " ")[0]
+	if testName != "" {
+		writeString = fmt.Sprintf("%s:[%s] [%s] %s\n",
+			entry.Time.Format("2006-01-02 15:04:05 -0700"), level, testName,
+			entry.Message)
+	} else {
+		writeString = fmt.Sprintf("%s:[%s] %s\n",
+			entry.Time.Format("2006-01-02 15:04:05 -0700"), level,
+			entry.Message)
+	}
+
+	b.WriteString(writeString)
 	return b.Bytes(), nil
 }
 
 func Fatalf(format string, args ...interface{}) {
-	dash.Fatal(format, args...)
-	tpLog.Errorf(format, args...)
+	pc, _, line, _ := runtime.Caller(1)
```

Comment:
@mkoppal-px , looks like these below 4 lines are repeated in all the methods in log.go file, shall we wrap together in a method named `extendedFormat` or something similar like that ? 
```  
        pc, _, line, _ := runtime.Caller(1)
	callerFuncSlice := strings.Split(runtime.FuncForPC(pc).Name(), "/")
	callerFunc := fmt.Sprintf("%s:#%d", callerFuncSlice[len(callerFuncSlice)-1], line)
	extendedFormat := fmt.Sprintf("[%s] - %s", callerFunc, format)
```

Reply:
@Madanagopal19 if we create another function, the caller function will change and it will not be representing accurately

Reply:
got it.